labelbox/data/annotation_types/data/tiled_image.py

```
@@ -0,0 +1,364 @@
+
+    If specified version is 2, converts bounds from [lng,lat] to [lat,lng]
+
+    Requires the following args:
+        tile_layer: TileLayer
+        tile_bounds: TiledBounds
+        zoom_levels: List[int]
+    Optional args:
+        max_native_zoom: int = None
+        tile_size: Optional[int]
+        version: int = 2
+        alternative_layers: List[TileLayer]     
+
+    >>> tiled_image_data = TiledImageData(tile_layer=TileLayer,
+                                  tile_bounds=TiledBounds,
+                                  zoom_levels=[1, 12])
+    """
+    tile_layer: TileLayer
+    tile_bounds: TiledBounds
+    alternative_layers: List[TileLayer] = None
+    zoom_levels: conlist(item_type=int, min_items=2, max_items=2)
+    max_native_zoom: int = None
+    tile_size: Optional[int] = TMS_TILE_SIZE
+    version: int = 2
+
+    def _as_raster(self, zoom=0) -> RasterData:
+        """Converts the tiled image asset into a RasterData object containing an
+        np.ndarray.
+
+        Uses the minimum zoom provided to render the image.
+        """
+        if self.tile_bounds.epsg == EPSG.SIMPLEPIXEL:
+            xstart, ystart, xend, yend = self._get_simple_image_params(zoom)
+
+        # Currently our editor doesn't support anything other than 3857.
+        # Since the user provided projection is ignored by the editor
+        #    we will ignore it here and assume that the projection is 3857.
+        else:
+            if self.tile_bounds.epsg != EPSG.EPSG3857:
+                logger.info(
+                    f"User provided EPSG is being ignored {self.tile_bounds.epsg}."
+                )
+            xstart, ystart, xend, yend = self._get_3857_image_params(zoom)
+
+        total_n_tiles = (yend - ystart + 1) * (xend - xstart + 1)
+        if total_n_tiles > MAX_TILES:
+            logger.info(
+                f"Too many tiles requested. Total tiles attempted {total_n_tiles}."
+            )
+            return None
```

Maybe move this into a separate function and raise an exception instead of logging the tile count.